sw/device/lib/dif/dif_base.c

```diff
@@ -0,0 +1,22 @@
+// Copyright lowRISC contributors.
+// Licensed under the Apache License, Version 2.0, see LICENSE for details.
+// SPDX-License-Identifier: Apache-2.0
+
+#include "sw/device/lib/dif/dif_base.h"
+
+#include <stdbool.h>
+
+bool toggle_to_bool(dif_toggle_t val) {
+  switch (val) {
+    case kDifToggleEnabled:
+      return true;
+    case kDifToggleDisabled:
+      return false;
+    default:
+      return false;
```

Comment:
Should this function return `dif_result_t` instead and have an out-param for the `bool`?

Reply:
Before, when this was a static helper function in many IP's DIF libs, it did not return a DIF return type (it just returned a bool for success or not). I'm happy to change that, since this has now become a shared helper in the dif_base lib, however, if I do, then this cannot be easily inlined when used (which loses some convenience and simplicity). Either format is fine with me, thoughts?

Reply:
With @mcy's [comment](https://github.com/lowRISC/opentitan/pull/9747#discussion_r774018899) above to add the `dif_is_valid_toggle()` function, it becomes pretty clean to inline of the `dif_toggle_to_bool()` and `dif_bool_to_toggle()` functions when used. 

However, the alternative would be to make these helper functions return a `dif_result_t` and check for valid toggle types in the helper functions themselves. Lmk which you would prefer!